packages/lb-annotation/src/core/pointCloud/render/index.ts

```diff
@@ -99,32 +105,48 @@ class PointCloudRender {
   // TODO, Just for showing.
   public generateNewPoints(segmentData: IPointCloudSegmentation) {
     const geometry = new THREE.BufferGeometry();
+    // get points by cover mode
+    const renderPoints =
+      this.store.segmentCoverMode === EPointCloudSegmentCoverMode.Cover
+        ? segmentData.points
+        : this.store.splitPointsFromPoints(segmentData.points, segmentData.coverPoints);
     // itemSize = 3 因为每个顶点都是一个三元组。
-    geometry.setAttribute('position', new THREE.BufferAttribute(segmentData.points, 3));
+    geometry.setAttribute('position', new THREE.BufferAttribute(renderPoints, 3));
 
-    const pointsMaterial = new THREE.PointsMaterial({ color: this.currentColor, size: 10 });
+    const pointsMaterial = new THREE.PointsMaterial({ color: this.getCurrentColor(segmentData.attribute), size: 10 });
     const newPoints = new THREE.Points(geometry, pointsMaterial);
     newPoints.name = segmentData.id;
 
     this.store.scene.add(newPoints);
     this.render3d();
   }
 
-  public updateNewPoints = (segmentData = this.store.cacheSegData) => {
+  public updateNewPoints = (segmentData: IPointCloudSegmentation) => {
     const originPoints = this.store.scene.getObjectByName(segmentData?.id ?? '') as THREE.Points;
+    const renderPoints =
```

Comment:
logic needs to be removed from the render. Render just get the data to draw

Reply:
done